File: mobile/src/main/java/com/windscribe/mobile/robert/RobertSettingsView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.robert;

import com.windscribe.mobile.adapter.RobertSettingsAdapter;

public interface RobertSettingsView {

    void hideProgress();

    void openUrl(String url);

    void setAdapter(RobertSettingsAdapter robertSettingsAdapter);

    void setTitle(String title);

    void setWebSessionLoading(boolean loading);

    void showError(String error);

    void showErrorDialog(String error);

    void showProgress();

    void showToast(String message);
}

File: mobile/src/main/java/com/windscribe/mobile/robert/RobertSettingsView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.robert

import com.windscribe.mobile.adapter.RobertSettingsAdapter

interface RobertSettingsView {
    fun hideProgress()
    fun openUrl(url: String)
    fun setAdapter(robertSettingsAdapter: RobertSettingsAdapter)
    fun setTitle(title: String)
    fun setWebSessionLoading(loading: Boolean)
    fun showError(error: String)
    fun showErrorDialog(error: String)
    fun showProgress()
    fun showToast(message: String)
}